django/db/backends/oracle/creation.py

```diff
@@ -99,6 +100,71 @@ def _create_test_db(self, verbosity=1, autoclobber=False, keepdb=False):
         self._switch_to_test_user(parameters)
         return self.connection.settings_dict['NAME']
 
+    def _clone_test_db(self, suffix, verbosity, keepdb):
+        def _clone_test_user(self, cursor, parameters, verbosity, keepdb):
+            try:
+                self._create_test_user(cursor, parameters, verbosity, keepdb)
+            except Exception as e:
+                # ORA-01920: User already exists
+                if 'ORA-01920' in str(e):
+                    self._destroy_test_user(cursor, parameters, verbosity)
+                    self._create_test_user(cursor, parameters, verbosity, keepdb)
+
+        def _execute_test_db_cloning(self, cursor, parameters, verbosity, keepdb):
+            try:
+                self._execute_test_db_creation(cursor, parameters, verbosity, keepdb)
+            except Exception as e:
+                # ORA-01543: Tablespace already exists
+                if 'ORA-01543' in str(e):
+                    self._execute_test_db_destruction(cursor, parameters, verbosity)
+                    self._execute_test_db_creation(cursor, parameters, verbosity, keepdb)
+
+        parameters = self._get_test_db_params(suffix)
+        main_test_user = self.connection.settings_dict['USER']
+        main_test_password = parameters['password'] = self.connection.settings_dict['PASSWORD']
+        self.connection.settings_dict['USER'] = self.connection.settings_dict['SAVED_USER']
+        self.connection.settings_dict['PASSWORD'] = self.connection.settings_dict['SAVED_PASSWORD']
+        main_login = self.connection._connect_string()
+        self.connection.settings_dict['USER'] = main_test_user
+        self.connection.settings_dict['PASSWORD'] = main_test_password
+        cloned_test_user = parameters['user']
+        with self._maindb_connection.cursor() as cursor:
+            _execute_test_db_cloning(self, cursor, parameters, verbosity, keepdb)
+            _clone_test_user(self, cursor, parameters, verbosity, keepdb)
+
+        if int(suffix) < 2:
+            exp_cmd = [
+                'expdp',
+                main_login,
+                'REUSE_DUMPFILES=Y',
+                f'schemas={main_test_user}',
+                f'dumpfile={main_test_user}.dmp',
+            ]
+            subprocess.run(exp_cmd, check=True, shell=False)
+        imp_cmd = [
+            'impdp',
+            main_login,
+            f'dumpfile={main_test_user}.dmp',
+            f'remap_schema={main_test_user}:{cloned_test_user}',
+        ]
+        subprocess.run(imp_cmd, shell=False)
```

Comment:
It's rather rare that folks have the Oracle database engine installed locally, so it crashes because `expdp` and `impdp` are not available.